### ISSUE TITLE
Ability to customise multiple selector controller

### ISSUE THREAD

**Issue by mkoppanen** [xmartlabs/XLForm#410]
At the moment XLFormRowDescriptorTypeMultipleSelector ignores viewController type completely and uses hardcoded XLFormOptionsViewController. We ran some user testing recently and the uniform opinion was that the way to do multiple selection was confusing. The users did not find it intuitive that back button actually sets the values.

Hence I want to customise the XLFormRowDescriptorTypeMultipleSelector to have "Done" button instead of the normal back button. Something like the following seems to do the trick:

````
diff --git a/XLForm/XL/Cell/XLFormSelectorCell.m b/XLForm/XL/Cell/XLFormSelectorCell.m
index 7a89ed3..20a16c9 100644
--- a/XLForm/XL/Cell/XLFormSelectorCell.m
+++ b/XLForm/XL/Cell/XLFormSelectorCell.m
@@ -194,7 +194,12 @@
             }
         }
         else if (self.rowDescriptor.selectorOptions){
-            XLFormOptionsViewController * optionsViewController = [[XLFormOptionsViewController alloc] initWithStyle:UITableViewStyleGrouped titleHeaderSection:nil titleFooterSection:nil];
+
+            XLFormOptionsViewController * optionsViewController = (XLFormOptionsViewController *)[self controllerToPresent];
+            if (!optionsViewController) {
+                optionsViewController = [[XLFormOptionsViewController alloc] initWithStyle:UITableViewStyleGrouped titleHeaderSection:nil titleFooterSection:nil];
+            }
+
             optionsViewController.rowDescriptor = self.rowDescriptor;
             optionsViewController.title = self.rowDescriptor.selectorTitle;
                        
@@ -217,7 +222,10 @@
     else if ([self.rowDescriptor.rowType isEqualToString:XLFormRowDescriptorTypeMultipleSelector] || [self.rowDescriptor.rowType isEqualToString:XLFormRowDescriptorTypeMultipleSelectorPopover])
     {
         NSAssert(self.rowDescriptor.selectorOptions, @"selectorOptions property shopuld not be nil");
-        XLFormOptionsViewController * optionsViewController = [[XLFormOptionsViewController alloc] initWithStyle:UITableViewStyleGrouped titleHeaderSection:nil titleFooterSection:nil];
+        XLFormOptionsViewController * optionsViewController = (XLFormOptionsViewController *)[self controllerToPresent];
+        if (!optionsViewController) {
+            optionsViewController = [[XLFormOptionsViewController alloc] initWithStyle:UITableViewStyleGrouped titleHeaderSection:nil titleFooterSection:nil];
+        }
         optionsViewController.rowDescriptor = self.rowDescriptor;
         optionsViewController.title = self.rowDescriptor.selectorTitle;
         

````

**Comment by memmons-ntst** [comment]
The choices seem to be:

1. No changes -- keep the back button
2. Add a done button, but remove the back button
3. Add a done button, keep the back button
4. Add a done button and a cancel button

Notes:

1. I agree it feels a little odd to have a multi-select with only a back button, but it's really not that bad.
2. I've never seen an Apple implementation of a multi-select that only has a done button. Feels very modal and not at all what a pushed view should feel like.
3. I've seen Apple do this once in their UI in the Music app on a modal form when creating a new playlist. The "done" in that case means dismiss the modal form, so this isn't very similar to what you are suggesting. Having a done button and a back button makes no sense for a pushed form.
4. This seems the most consistent of the options but also feels very modal and would require maintaining state if the cancel button is tapped. The added complexity seems a poor trade-off for the currently simple UI.

tl;dr Leaving the UI as-is with a back button for multi-select values seems the least wonky of the UI solutions.

**Comment by cerupcat** [comment]
Option 5. Add a "Save" button to the top right. Save, pops the view controller and saves, back only pushes back (doesn't save).

I agree that "done" doesn't necessarily make sense. It's more a question whether people know that they're saving when selecting items and that hitting the back button will save their choice. It's more explicit to instead have a "save" and back button IMO.

**Comment by RyanBertrand** [comment]
I set the back button title to "Save". That way we don't have to add an additional button and it just works.

**Comment by cerupcat** [comment]
@RyanBertrand that's a good way to do it as long as you don't want a 'cancel' button.

If you want 'cancel' and 'save' I'd recommend just subclassing XLFormOptionsViewController and that way you can add override the behavior you want. This is what I'm learning towards for now.

**Comment by mats-claassen** [comment]
I believe this decision is up to every developer so that subclassing XLFormOptionsViewController is the best solution
